From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: mohammed jasem alaajel <xrambad@gmail.com>
Date: Wed, 19 Apr 2023 09:41:08 +0400
Subject: [PATCH] Add AttemptItemMergeEvent

Basically this event fires before ItemMergeEvent which allows see what item source item want to merge with and can be canceld and class is exact copy of ItemMergeEvent

diff --git a/src/main/java/io/papermc/paper/event/entity/AttemptItemMergeEvent.java b/src/main/java/io/papermc/paper/event/entity/AttemptItemMergeEvent.java
new file mode 100644
index 0000000000000000000000000000000000000000..9ca917a1a4e4bf3ea3250bbcaf87650ce56c8008
--- /dev/null
+++ b/src/main/java/io/papermc/paper/event/entity/AttemptItemMergeEvent.java
@@ -0,0 +1,59 @@
+package io.papermc.paper.event.entity;
+
+import org.bukkit.entity.Item;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.entity.EntityEvent;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * thrown when are Items trying to merge
+ */
+public class AttemptItemMergeEvent extends EntityEvent implements Cancellable {
+
+    private static final HandlerList handlers = new HandlerList();
+    private boolean cancelled;
+    private final Item target;
+
+    public AttemptItemMergeEvent(@NotNull Item item, @NotNull Item target) {
+        super(item);
+        this.target = target;
+    }
+
+    @Override
+    public boolean isCancelled() {
+        return cancelled;
+    }
+
+    @Override
+    public void setCancelled(boolean cancelled) {
+        this.cancelled = cancelled;
+    }
+
+    @NotNull
+    @Override
+    public Item getEntity() {
+        return (Item) entity;
+    }
+
+    /**
+     * Gets the Item entity the main Item is being merged into.
+     *
+     * @return The Item being merged with
+     */
+    @NotNull
+    public Item getTarget() {
+        return target;
+    }
+
+    @NotNull
+    @Override
+    public HandlerList getHandlers() {
+        return handlers;
+    }
+
+    @NotNull
+    public static HandlerList getHandlerList() {
+        return handlers;
+    }
+}
